File: JavaScript/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    const counts = new Map();
    array.forEach(a => {
        counts.set(a, (counts.get(a) || 0) + 1);
    });
    
    let a_num = 0;
    let maxCount = 0;
    
    counts.forEach((value, key) => {
        if (value == maxCount) a_num = -1;
        if (value > maxCount) {
            a_num = key;
            maxCount = value;
        }
    })
    
    const answer = a_num;
    return answer;
}

File: Python3/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
from collections import Counter

def solution(array):
    counts = Counter(array)
    
    answer = 0
    max_count = 0
    
    for key, value in counts.items():
        if value == max_count:
            answer = -1
        if value > max_count:
            answer = key
            max_count = value
    
    return answer